### PR TITLE
Add fyle camera in my-expenses page

### PR DESCRIPTION
**Pull request by yksolanki9**
https://user-images.githubusercontent.com/39493102/154623706-a1af563c-11f0-4c86-9be1-572bd52d7901.mp4

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -38,4 +38,5 @@ ios/App/App/GoogleService-Info.plist
 ios/App/App/Info.plist
 package-lock.json
 setup.sh
-.env
\ No newline at end of file
+.env
+.angular
```
Comment (Dimple16) [inline .gitignore:42]: is this an intentional change?

File: src/app/shared/components/expenses-card/expenses-card.component.ts [hunk 2]
```diff
@@ -105,7 +115,9 @@ export class ExpensesCardComponent implements OnInit {
     private fileService: FileService,
     private popoverController: PopoverController,
     private networkService: NetworkService,
-    private transactionOutboxService: TransactionsOutboxService
+    private transactionOutboxService: TransactionsOutboxService,
+    private modalController: ModalController,
+    public platform: Platform
```
Comment (Dimple16) [inline src/app/shared/components/expenses-card/expenses-card.component.ts:120]: use this as private and add a variable to use in the HTML

File: src/app/shared/components/expenses-card/expenses-card.component.html [hunk 3]
```diff
@@ -28,6 +28,13 @@
           *ngIf="!isOutboxExpense; else outboxExpenseDisplay"
           (click)="addAttachments($event)"
         >
+          <input
+            *ngIf="platform.is('ios')"
```
Comment (Dimple16) [inline src/app/shared/components/expenses-card/expenses-card.component.html:32]: this will change according to the below comment

**Review by Dimple16 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (yksolanki9) [reply .gitignore:42]: Yes

File: src/app/shared/components/expenses-card/expenses-card.component.ts (hunk 2 above)
Comment (yksolanki9) [reply src/app/shared/components/expenses-card/expenses-card.component.ts]: Done

**Review by Dimple16 — APPROVED** [review]

File: src/app/fyle/add-edit-expense/add-edit-expense.page.html [hunk 4]
```diff
@@ -102,13 +102,7 @@
...
                 class="add-edit-expense--receipt-image"
               >
-                <input
-                  *ngIf="platform.is('ios')"
-                  type="file"
-                  #fileUpload
-                  accept="application/pdf,image/*"
-                  style="display: none"
-                />
+                <input *ngIf="isIos" type="file" #fileUpload accept="application/pdf,image/*" style="display: none" />
```
Comment (Julias0) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.html:105]: dei no inline styling. Please remove this

File: src/app/fyle/add-edit-expense/add-edit-expense.page.html [hunk 5]
```diff
@@ -127,13 +121,7 @@
...
                   add_circle
                 </mat-icon>
-                <input
-                  *ngIf="platform.is('ios')"
-                  type="file"
-                  #fileUpload
-                  accept="application/pdf,image/*"
-                  style="display: none"
-                />
+                <input *ngIf="isIos" type="file" #fileUpload accept="application/pdf,image/*" style="display: none" />
```
Comment (Julias0) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.html:124]: please remove the inline styling from everywhere

File: src/app/shared/components/expenses-card/expenses-card.component.ts [hunk 6]
```diff
@@ -329,63 +345,115 @@ export class ExpensesCardComponent implements OnInit {
...
 
+  attachReceipt(receiptDetails: ReceiptDetail) {
+    this.attachmentUploadInProgress = true;
+    const attachmentType = this.fileService.getAttachmentType(receiptDetails.type);
+
+    this.inlineReceiptDataUrl = attachmentType !== 'pdf' && receiptDetails.dataUrl;
+    from(this.transactionOutboxService.fileUpload(receiptDetails.dataUrl, attachmentType))
+      .pipe(
+        tap((fileObj: FileObject) => {
+          this.expense.tx_file_ids = [];
```
Comment (Julias0) [inline src/app/shared/components/expenses-card/expenses-card.component.ts:427]: extract out this part into a separate function

File: src/app/shared/components/expenses-card/expenses-card.component.ts [hunk 7]
```diff
@@ -329,63 +345,115 @@ export class ExpensesCardComponent implements OnInit {
...
+          this.expense.tx_file_ids.push(fileObj.id);
```
Comment (Julias0) [inline src/app/shared/components/expenses-card/expenses-card.component.ts:428]: any way we can avoid doing this here and do this inside the subscribe?

File: src/app/shared/components/expenses-card/expenses-card.component.ts (hunk 6 above)
Comment (yksolanki9) [reply src/app/shared/components/expenses-card/expenses-card.component.ts]: Made a separate function `setThumnail()` for this

**Review by Julias0 — APPROVED** [review]

**Pull request merged**